apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ir/Ir.kt

```diff
@@ -169,21 +170,26 @@ data class IrModel(
 
 /**
  * @param condition a condition for reading the property
- * @param requiresBuffering true if this property contains synthetic properties
- * @param hidden allows to hide a property from the model but still have it part of the selections.
- * This is used for typename in compat models because the adapters need to read __typename
+ * @param requiresBuffering true if this property contains synthetic properties and needs to be buffered
  */
 data class IrProperty(
     val info: IrFieldInfo,
     val override: Boolean,
     val condition: BooleanExpression<BTerm>,
-    val requiresBuffering: Boolean,
-    val hidden: Boolean,
```

Comment:
Nice side effect of not relying on `IrProperty` to get the __typename in adapters, we don't need this `hidden` hack to hide `__typename` from the compat `Fragments` model